packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1136,6 +1146,38 @@ export class KeyringController extends BaseController<
     return await keyring.signMessage(address, messageParams.data);
   }
 
+  /**
+   * Signs EIP-7702 Authorization message by calling down into a specific keyring.
+   *
+   * @param params - EIP7702AuthorizationParams object to sign.
+   * @returns Promise resolving to an EIP-7702 Authorization signature.
+   * @throws Will throw UnsupportedSignEIP7702Authorization if the keyring does not support signing EIP-7702 Authorization messages.
+   */
+  async signEip7702Authorization(
+    params: Eip7702AuthorizationParams,
+  ): Promise<string> {
+    const from = ethNormalize(params.from) as Hex;
+
+    const keyring = (await this.getKeyringForAccount(from)) as EthKeyring<Json>;
+
+    if (!keyring.signEip7702Authorization) {
+      throw new Error(
+        KeyringControllerError.UnsupportedSignEip7702Authorization,
+      );
+    }
+
+    const { chainId, nonce } = params;
+    const contractAddress = ethNormalize(params.contractAddress) as
+      | Hex
+      | undefined;
+
+    return await keyring.signEip7702Authorization(from, [
+      chainId,
+      contractAddress as Hex,
```

Comment:
Perhaps this is what you meant to do:

```suggestion
    const contractAddress = ethNormalize(params.contractAddress) as Hex;

    return await keyring.signEip7702Authorization(from, [
      chainId,
      contractAddress,
```

Ideally we'd validate the response from `ethNormalize` is non-falsy as well, or use a normalization function that can't return `undefined`. But I see we make this assumption elsewhere already, so we can leave this improvement for a future PR.